### ISSUE TITLE
crash when deleting a device

### ISSUE THREAD

**Issue by aliefhooghe** [ossia/score#833]
This crash happen for osc_query device, joystick and wiimote. I haven't tested other protocol at all, and thus maybe other device type have the same issue.

The crash can be reproduced by following theses instructions : 
- Create the device
- Unfold the parameter tree at least one time (without this, the crash doesn't happen)
- Remove the device

This will cause a deadlock exception.

Note than I successfully reproduced this behavior with a dummy device which just instantiate some parameters.

**Comment by jcelerier** [comment]
so, for the record, this is what was happening : (to understand I had to put a breakpoint on the throwing of exceptions - I did it through my IDE but in gdb / lldb it's a simple `catch throw`).

- the device is removed: `generic_device::~generic_device()`
- the mutex at generic_device.cpp:36 is locked because the device structure is being changed
- childrens are being removed recursively, every time setting up a lock
- at some point, a node is removed and calls `generic_node::remove_parameter()` which in turn does ` m_device.on_parameter_removing(*addr);`. this is a callback handler, and here of course the great rule of threading is broken : never call user-provided code (e.g. callbacks) while mutexes are being locked.
- the user-provided code, e.g. score, calls the score function: `DeviceInterface::addressRemoved(const ossia::net::parameter_base& addr)` which in turns calls another function (a bit further down the stack trace) : `findNodeFromPath`. This one calls back to libossia: `node_base::find_child(const QString& name)`... which locks the mutex again. Locking a mutex twice in a thread is not allowed and throws an exception.
- remember that we are in the destructor of `generic_device`: throwing an exception from a destructor in C++ is not allowed -> std::terminate is called
